python/NEXT.md

```diff
@@ -0,0 +1,41 @@
+# Next
+Some thoughts on the next steps for refactoring.
+
+In general there should an iterative refactoring of legacy updater. In which Item type should be
+refactored one by one.
+
+## Side effects
+The main interface for updating uses side effects. Meaning that the update doesn't return a new list but updates the old list.
+
+In my experience side effects should be avoided as they make code bases harder to understand and test.
+
+It might then be an idea the update be changed so that it returns a new list of Items.
+
+
+## ItemDefinition
+The side effect of updating `sell_in` property to define the date makes me feel a bit uncomfortable. Firstly because of the case:
+
+- Updates are done on a daily base
+- An undetected error in updater is committed and used for a number of days
+- The error is then detected and each Item has to then be fixed by hand
+- This could be extremely time consuming if the list of items is large
```

Comment:
Nit: This could be spelled out more precisely.

Reply:
https://github.com/benjamincerigo/GildedRose-Refactoring-Kata/pull/1/commits/9e83987e14ecfdf66984124043965d4bdabfd8c2